src/test/java/net/kaczmarzyk/RequestBodyHandlingE2eTest.java

```diff
@@ -111,6 +111,30 @@ public void findsByIdProvidedInRequestBodyWhenContentTypeContainsParameter() thr
                .andExpect(jsonPath("$[0].firstName").value(homerSimpson.getFirstName()));
     }
 
+    @Test
+    public void findsByIdProvidedInRequestBodyWhenContentTypeContainsCharsetParameter() throws Exception {
+        mockMvc.perform(post("/customers/search")
+                        .accept(MediaType.APPLICATION_JSON)
+                        .contentType(MediaType.APPLICATION_JSON_VALUE + ";charset=UTF-8")
+                        .content(" { \"customerId\": \"" + homerSimpson.getId() + "\" }"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$.length()").value(1))
+                .andExpect(jsonPath("$[0].firstName").value(homerSimpson.getFirstName()));
+    }
+
+    @Test
+    public void findsByIdProvidedInRequestBodyWhenContentTypeContainsBoundaryParameter() throws Exception {
```

Comment:
Consider adding the test case for the scenario in which the content-type param is not present at all.